addons/account/models/sequence_mixin.py

```diff
@@ -18,7 +18,7 @@ class SequenceMixin(models.AbstractModel):
     _sequence_field = "name"
     _sequence_monthly_regex = r'^(?P<prefix1>.*?)(?P<year>\d{4})(?P<prefix2>\D*?)(?P<month>\d{2})(?P<prefix3>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
     _sequence_yearly_regex = r'^(?P<prefix1>.*?)(?P<year>\d{4})(?P<prefix2>\D+?)(?P<seq>\d*)(?P<suffix>\D*?)$'
-    _sequence_fixed_regex = r'^(?P<prefix1>.*?)(?P<seq>\d*)(?P<suffix>\D*?)$'
+    _sequence_fixed_regex = r'^(?P<prefix1>.*?)(?P<seq>\d+)(?P<suffix>\D*?)$'
```

Comment:
This should be done in monthly/yearly regexes too.
Note that if someone uses `sequence_override_regex`, you are not guaranteed to have `(?P<seq>\d+)`
https://github.com/odoo/odoo/blob/0c67025c40c0d0ac86c19638addc999e4725418f/addons/account/models/account_move.py#L40-L50

It means the constraint `_constrains_regex_match` can be bypassed.

Reply:
By the way, you could move the name `seq` from `optional` to `required` in `_check_grouping` in `_deduce_sequence_number_reset` to have the same behaviour (maybe needs to reword the validation error)